contracts/verifiers/UniversalVerifier.sol

```diff
@@ -21,216 +23,234 @@ contract UniversalVerifier is
-    /// @param requestId The ID of the ZKP request
-    /// @param request The ZKP request data
-    function setZKPRequest(
-        uint64 requestId,
-        IZKPVerifier.ZKPRequest calldata request
-    ) public override(RequestOwnership, ValidatorWhitelist, ZKPVerifierBase) {
-        super.setZKPRequest(requestId, request);
-
-        emit ZKPRequestSet(
-            requestId,
-            _msgSender(),
-            request.metadata,
-            address(request.validator),
-            request.data
-        );
-    }
-
     /**
-     * @dev Set the list of ZKP requests for the list of requestIds in the same order.
-     * @param requestIds Request ids of the ZKP requests.
-     * @param requests ZKP requests to set.
+     * @dev Sets an auth type
+     * @param authMethod The auth type to add
      */
-    function setZKPRequests(
-        uint64[] calldata requestIds,
-        ZKPRequest[] calldata requests
-    ) public override(RequestOwnership, ValidatorWhitelist, ZKPVerifierBase) {
-        for (uint256 i = 0; i < requestIds.length; i++) {
-            setZKPRequest(requestIds[i], requests[i]);
-        }
+    function setAuthMethod(IVerifier.AuthMethod calldata authMethod) public override onlyOwner {
+        super.setAuthMethod(authMethod);
+        emit AuthMethodSet(authMethod.authMethod, address(authMethod.validator), authMethod.params);
     }
 
-    /// @dev Update a ZKP request
-    /// @param requestId The ID of the ZKP request
-    /// @param request The ZKP request data
-    function updateZKPRequest(
-        uint64 requestId,
-        IZKPVerifier.ZKPRequest calldata request
-    ) public onlyOwner {
-        super._updateZKPRequest(requestId, request);
+    /**
+     * @dev Updates a request
+     * @param request The request data
+     */
+    function updateRequest(IVerifier.Request calldata request) public onlyOwner {
+        super._updateRequest(request);
 
-        emit ZKPRequestUpdate(
-            requestId,
+        emit RequestUpdate(
+            request.requestId,
             _msgSender(),
             request.metadata,
             address(request.validator),
-            request.data
+            request.params
         );
     }
 
-    /// @dev Submits a ZKP response and updates proof status
-    /// @param requestId The ID of the ZKP request
-    /// @param inputs The input data for the proof
-    /// @param a The first component of the proof
-    /// @param b The second component of the proof
-    /// @param c The third component of the proof
-    function submitZKPResponse(
-        uint64 requestId,
-        uint256[] memory inputs,
-        uint256[2] memory a,
-        uint256[2][2] memory b,
-        uint256[2] memory c
-    ) public override {
-        super.submitZKPResponse(requestId, inputs, a, b, c);
-        emit ZKPResponseSubmitted(requestId, _msgSender());
+    /**
+     * @dev Sets a multiRequest
+     * @param multiRequest The multiRequest data
+     */
+    function setMultiRequest(
```

Comment:
Does it check for validator whitelist, disabled requests. Does it set request owner?